guide/creating-your-bot/parsing-mention-arguments.md

```diff
@@ -0,0 +1,219 @@
+
+```
+!ban Because he was rude to @Victim
+```
+
+The bot will still ban someone, but it will be the @Victim again. `message.mentions.users` still contains a mention, which the bot will use. But in reality you would want your bot to be able to tell the user he used the command incorrectly.
+
+### How Discord mentions work
+
+Discord uses a special syntax to embed mentions in a message. For user mentions it is the user's ID with `<@` at the start and `>` at the end, like this: `<@86890631690977280>`. If they have a nickname there will also be a be a `!` after the `@`.  
+Role mentions and channel mentions work similarly. Role mentions look like `<@&134362454976102401>` and channel mentions like `<#222197033908436994>`.
+
+That means when you receive a message from the Discord API and it contains mentions the message's content will contain that special syntax.  
+If you send
+
+<discord-messages>
+	<discord-message author="User" avatar="djs">
+		I think we should add @GoodPerson to the @Mod role.
+	</discord-message>
+</discord-messages>
+
+then the `message.content` for that message will look something like this
+
+<!-- eslint-skip -->
+```js
+'I think we should add <@86890631690977280> to the <@&134362454976102401> role.'
+```
+
+### Implementation
+
+So, how do you actually use this new information for your bot?  
+Most of your code will not change, however instead of using `message.mentions` to find the mentioned users you will have to do it manually.  
+This may sound scary at first, but once you see the code you will see it is actually pretty simple.
+
+Say you already have a simple command handler like this:
+
+```js
+client.on('message', message => {
+	// Don not go any further if there is no command prefix.
+	if (!message.content.startsWith(config.prefix)) return;
+
+	// Remove the prefix from the message.
+	const withoutPrefix = message.content.slice(config.prefix.length);
+	// Split the message by spaces
+	const split = withoutPrefix.split(/ +/);
+	// The first element is the command, the rest are the arguments.
+	const command = split[0];
+	const args = split.slice(1);
+
+	// ....Some commands here....
+});
+```
+
+Now you can easily test the waters by upgrading the avatar command from last time.
+This is what we have so far. It is pretty simple, it will show the avatar of who used the command.
+
+```js
+if (command === 'avatar') {
+	// What to put here to use the mention?
+	const user = message.author;
+
+	return message.channel.send(`${user.username}'s avatar: ${user.displayAvatarURL}`);
+}
+```
+
+But how do you actually get the correct user now? Well, this requires a few simple steps.  
+Putting it into a function will make it easily reusable. We will use the name `getUserFromMention` here.
+
+```js
+function getUserFromMention(mention) {
+	// Check if `mention` contains a string.
+	// If not, just return undefined to signal there is no user to be found.
+	if (!mention) return undefined;
```

Comment:
Same for this area; if you read the code, you'll understand what's going on faster than if you would read the comments.

Reply:
Also not sure if we should be using `return undefined`, just `return`, or `return null`. For some reason I feel like null is more appropriate, but it's not a big deal either way.